integration/log_integration_test.go

```diff
@@ -310,10 +311,15 @@ func readbackLogEntries(logID int64, client trillian.TrillianLogClient, params t
 
 			delete(leafDataPresenceMap, string(leaf.LeafValue))
 
-			hash := hasher.HashLeaf(response.Leaves[l].LeafValue)
+			hash := hasher.HashLeaf(leaf.LeafValue)
 
 			if got, want := hex.EncodeToString(hash), hex.EncodeToString(leaf.MerkleLeafHash); got != want {
-				return nil, fmt.Errorf("leaf hash mismatch expected got: %s want: %s", got, want)
+				return nil, fmt.Errorf("leaf %d hash mismatch expected got: %s want: %s", leaf.LeafIndex, got, want)
+			}
+
+			// Ensure that the ExtraData in the leaf made it through the roundtrip
```

Comment:
(nit) be nice to comment the convention that data = "Leaf %d", extra = "Extra %d"